spi/control-plane/contract-spi/src/main/java/org/eclipse/edc/connector/controlplane/contract/spi/types/offer/ContractOffer.java

```diff
@@ -103,8 +104,17 @@ public Builder policy(Policy policy) {
 
         public ContractOffer build() {
             Objects.requireNonNull(contractOffer.id, "Id must not be null");
+            var offerId = ContractOfferId.parseId(contractOffer.id)
+                    .orElseThrow(failure -> new IllegalArgumentException(failure.getFailureDetail()));
```

Comment:
same here, this validation would break for every contract offer created by a connector that's not an EDC.